src/selection.js

```diff
@@ -352,3 +372,92 @@ export default class Selection extends Cursor {
     this.setVisibleSelection()
   }
 }
+
+function findStartExcludingWhitespace ({root, startContainer, startOffset, whitespacesOnTheLeft}) {
+  const isTextNode = startContainer.nodeType === textNode
+  if (!isTextNode) {
+    return findStartExcludingWhitespace({
+      root,
+      startContainer: startContainer.childNodes[startOffset],
+      startOffset: 0,
+      whitespacesOnTheLeft
+    })
+  }
+
+  const offsetAfterWhitespace = startOffset + whitespacesOnTheLeft
+  if (startContainer.length > offsetAfterWhitespace) {
+    return [startContainer, offsetAfterWhitespace]
+  }
+
+  // *** Start hack ***
+  // Pass the root so that the iterator can traverse to siblings
+  const iterator = new NodeIterator(root)
+  // Set the position to the node which is selected
+  iterator.nextNode = startContainer
+  // Iterate once to avoid returning self
+  iterator.getNextTextNode()
+  // *** End hack ***
```

Comment:
@peyerluk please could you take a look at this "hack". In order for this to work I need to set the root element in the `NodeIterator` as the range `commonAncestorContainer`, but the first node should be the range `startContainer`. Should I modify `NodeIterator` to accept different values for both, or keep the "hack" as it is?

Also, it always return the `startContainer` when I call `getNextTextNode()` for the first time (because at this point in the code `startContainer` will be a text node. It looks like this is a deliberate design consideration, so are you happy enough with the "unused" `iterator.getNextTextNode()` call in this code?

Reply:
For this case it is best you add enough tests for a few different use cases. Then you don't have to worry about this 'hack' too much.

But code like this is an ideal candidate for test driven development.

Also you should move the helper functions out of the selection class.

Reply:
I've added a couple of tests and moved the helper functions.